website/src/_posts/2022-09-3.0.md

```diff
@@ -0,0 +1,146 @@
+
+But, it didn’t come without its own set of new problems:
+
+* It tries to do the exact same, but it looks like a different product.
+* It’s confusing for users whether they want to use Robodog or Uppy directly.
+* Robodog is more ergonomic because it’s limited. When you hit such a limit, you
+  need to refactor everything to Uppy with plugins.
+
+This has now led us to deprecating Robodog and embrace Uppy for its strong suits: modularity and flexibility. At the same time, we also introduced something to take away some repetitive integration code: [`@uppy/remote-sources`](/docs/remote-sources).
+
+To mimic the Robodog implementation with all its features, you can use the code snippet below. But chances are Robodog did more than you need so feel free to remove things or go through the [list of plugins](/docs/plugins/) and install and use the ones you need.
+
+You can also checkout how we migrated the Robodog example ourselves in this [commit](https://github.com/transloadit/uppy/commit/089aaed615c77bafaf905e291b6b4e82aaeb2f6f).
+
+```js
+// Import just the plugins you need
+import Uppy from '@uppy/core'
+import Dashboard from '@uppy/dashboard'
+import RemoteSources from '@uppy/remote-sources'
+import Webcam from '@uppy/webcam'
+import ScreenCapture from '@uppy/screen-capture'
+import ImageEditor from '@uppy/image-editor'
+import Audio from '@uppy/audio'
+import Transloadit, { COMPANION_URL, COMPANION_ALLOWED_HOSTS } from '@uppy/transloadit'
+
+// Import their styles
+import '@uppy/core/dist/style.css'
+import '@uppy/dashboard/dist/style.css'
+import '@uppy/webcam/dist/style.css'
+import '@uppy/audio/dist/style.css'
+import '@uppy/screen-capture/dist/style.css'
+import '@uppy/image-editor/dist/style.css'
+
+// Initialize Uppy
+const uppy = new Uppy()
+  .use(Dashboard, { inline: true, target: '#app' })
+  .use(RemoteSources, { companionUrl: COMPANION_URL, companionAllowedHosts: COMPANION_ALLOWED_HOSTS })
+  .use(Webcam, { target: Dashboard })
+  .use(Audio, { target: Dashboard })
+  .use(ScreenCapture, { target: Dashboard })
+  .use(ImageEditor, { target: Dashboard })
+  .use(Transloadit, {
+    async getAssemblyOptions (file) {
+      // This is where you configure your auth key, auth secret, and template ID
+      // https://uppy.io/docs/transloadit/#getAssemblyOptions-file
+      //
+      // It is important to set the secret in production:
+      // https://transloadit.com/docs/topics/signature-authentication/
+      const response = await fetch('/some-endpoint')
+      return response.json()
+    },
+  })
+```
+
+## Companion 4.0
+
+### Streaming upload
+
+Streaming upload can now be enabled in Companion with Tus. This gives greatly improved upload speeds when using remote Providers and allows uploading potentially hundreds of gigabytes without needing a large server storage.
```

Comment:
I think it would be nice to add one sentence on why it only works with Tus?

Reply:
Ah, I will make it more clear that it now **also** works with Tus. In the previous version it worked with all destinations except Tus.

```suggestion
Streaming upload can now be enabled in Companion also when using Tus. This gives greatly improved upload speeds when using remote Providers and allows uploading potentially hundreds of gigabytes without needing a large server storage. Note that this feature was also available before Companion 4, but did not work with Tus.
```